### PR TITLE
add filter for flip auctions and make sure tend works

### PR DESCRIPTION
**Pull request by b-pmcg**
ID clashes would mean we need to add a filter for ilk type, but `tend` works again.

### CODE REVIEW

File: stores/auctionsStore.js [hunk 1]
```diff
@@ -4,14 +4,17 @@ import { AUCTION_DATA_FETCHER } from '../constants';
...
-  console.log('fetched events')
+  console.log('fetched events');
   await Promise.all(
     Object.keys(groupedEvents).map(async id => {
       try {
-        const { end, tic } = await service.getFlopDuration(id);
+        const { end, tic } =
+          type === 'flip'
+            ? await service.getFlipDuration(id, ilk)
+            : await service.getFlopDuration(id);
```
Comment (adrianleb) [inline stores/auctionsStore.js]: Does this maybe justify a `getDuration` convenience method within the service that internally calls `getFlipDuration` or `getFlopDuration`?

File: stores/auctionsStore.js [hunk 2]
```diff
@@ -263,6 +266,22 @@ const [useAuctionsStore, updateState] = create((set, get) => ({
...
+    const auctions = (await service.fetchFlipAuctions()).filter(
+      x => x.ilk === ilk
+    );
+    const transformedAuctions = await transformEvents(
+      auctions,
+      service,
+      'flip',
+      ilk
+    );
+    set({ auctions: transformedAuctions });
```
Comment (adrianleb) [inline stores/auctionsStore.js]: Can we store flip auctions under a different key maybe? User should be able to navigate between pages and this store will hold mixed data...

Comment (b-pmcg) [reply stores/auctionsStore.js]: Yeah, that makes sense. One issue with flip auctions is that each ilk will be numbered sequentially too, so there will be ID clashes which causes incorrect merging of ETH & BAT events with the same auctionId key. Would it help to have separate keys for each ilk, or should we handle that with a filter?

File: stores/auctionsStore.js (hunk 1 above)
Comment (b-pmcg) [reply stores/auctionsStore.js]: yep, much cleaner

File: hooks/useAuctionActions.js [hunk 3]
```diff
@@ -5,16 +5,27 @@ const useAuctionActions = () => {
   const { maker, web3Connected } = useMaker();
 
   //ETH tend
-  async function callTend(auctionId, lotSize, bidAmount) {
+  function callEthTend(auctionId, lotSize, bidAmount) {
```
Comment (peculiarity) [inline hooks/useAuctionActions.js]: Probably a single method that might accept `({auctionId, type}, lotSize, bidAmount)` or something like this.
Base on the `type` you can call the `flipEth` or `flipBat`.

What do you think about the following:

```javascript
function callTend(auctionid, colType, lotSize, bidAmount) {
  switch(colType) {
   case COLLATERAL_TYPES.ETH_A: return maker .service(AUCTION_DATA_FETCHER)....;
   case COLLATERAL_TYPES.BAT_A":return maker.service(AICTION_DATA_FETCHER)...;
   default throw Error("Unknown Collateral");
  }
}`

Comment (peculiarity) [reply hooks/useAuctionActions.js]: In a perfect world we will need a `factory method` that will return a `tendCall` based on the collateral type.

Comment (b-pmcg) [reply hooks/useAuctionActions.js]: yes good call, tried to simplify them a bit here.

File: hooks/useAllowances.js [hunk 4]
```diff
@@ -61,7 +84,10 @@ const useAllowances = () => {
         .service('smartContract')
         .getContract('MCD_VAT')
         .can(maker.currentAddress(), joinDaiAdapterAddress);
-      setHasJoinDaiHope(can.toNumber() === 1 ? true : false);
+      setHasHope(state => ({
+        ...state,
+        [MCD_JOIN_DAI]: can.toNumber() === 1 ? true : false
```
Comment (peculiarity) [inline hooks/useAllowances.js:89]: o.O Isn't `can.toNumber === 1` enough to return `true` or `false` ?

File: hooks/useAllowances.js [hunk 5]
```diff
@@ -74,7 +100,10 @@ const useAllowances = () => {
         .service('smartContract')
         .getContract('MCD_VAT')
         .can(maker.currentAddress(), flopAddress);
-      setHasFlopHope(can.toNumber() === 1 ? true : false);
+      setHasHope(state => ({
+        ...state,
+        [MCD_FLOP]: can.toNumber() === 1 ? true : false
```
Comment (peculiarity) [inline hooks/useAllowances.js:105]: Same like the comment above

File: hooks/useBalances.js [hunk 6]
```diff
@@ -23,17 +25,25 @@ const useBalances = () => {
   const fetchDaiBalance = () => {
     return maker.getToken(daiSymbol).balance();
   };
+  const fetchBatBalance = () => {
```
Comment (peculiarity) [inline hooks/useBalances.js:28]: Why don't we have `fetchBalanceOf = (token)=> ...`

**Pull request merged**